pkg/controllers/provisioning/provisioner.go

```diff
@@ -89,28 +71,44 @@ type Provisioner struct {
 }
 
 // Add a pod to the provisioner and block until it's processed. The caller
-// is responsible for verifying that the pod was scheduled correctly. In the
-// future, this may be expanded to include concepts such as retriable errors.
-func (p *Provisioner) Add(ctx context.Context, pod *v1.Pod) {
-	select {
-	case p.pods <- pod: // Block until pod is enqueued
-		<-p.wait
-	case <-p.running.Done(): // Leave if closed
+// is responsible for verifying that the pod was scheduled correctly.
+func (p *Provisioner) Add(ctx context.Context, pod *v1.Pod) <-chan struct{} {
+	p.RLock()
+	defer p.RUnlock()
+	p.batch.Add(pod)
```

Comment:
it might make sense to have batch.Add return the Done chan. It would save a line and you probably want to monitor the batch completion if you're adding something.